pkg/controller/dual-pods/server-requesting.go

```diff
@@ -0,0 +1,231 @@
+	}
+	logger.V(5).Info("Querying accelerators", "ip", ip, "port", port)
+	url := fmt.Sprintf("http://%s:%s%s", ip, port, stubapi.AcceleratorQueryPath)
+	gpuUUIDs, err := getGPUUUIDs(url)
+	if err != nil {
+		logger.V(5).Info("Not able to get GPU UUIDs at this time", "url", url, "error", err)
+		return err, true
+	}
+	if len(gpuUUIDs) == 0 {
+		logger.V(5).Info("No GPUs found for Pod", "name", requestingPod.Name)
+		return nil, true
+	}
+	logger.V(5).Info("Found GPUs for Pod", "name", requestingPod.Name, "gpuUUIDs", gpuUUIDs)
+	gpuIndices := mapToGPUIndices(gpuUUIDs)
+
+	// use the server patch to build the server-running pod
+	logger.V(5).Info("Building server-running pod from patch", "name", requestingPod.Name, "patch", serverPatch)
+	serverRunningPod, err := composeServerRunningPod(requestingPod, api.RunnerData{
+		GPUIndices: gpuIndices,
+		// TODO: these should be exposed as command-line flags or envars
+		LocalVolume:  "vcp-local-ip-172-31-58-228",
+		SharedVolume: "vcp-shared",
+	})
+	if err != nil {
+		logger.Error(err, "Failed to build server-running pod from patch", "name", requestingPod.Name, "patch", serverPatch)
+		return err, true
+	}
+
+	got, err := ctl.podLister.Pods(serverRunningPod.Namespace).Get(serverRunningPod.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.Error(err, "Failed to get existing server-running pod", "name", serverRunningPod.Name)
+		return err, true
+	}
+	if got != nil {
+		logger.V(5).Info("Server-running pod exists", "name", serverRunningPod.Name)
+
+		// TODO: we should reconcile the existing server-running pod with the one we just built
+		return nil, false
+	}
+
+	logger.V(2).Info("Creating server-running pod", "name", serverRunningPod.Name, "namespace", serverRunningPod.Namespace)
+	_, err = ctl.coreclient.Pods(serverRunningPod.Namespace).Create(ctx, serverRunningPod, metav1.CreateOptions{})
+	if err != nil {
+		logger.Error(err, "Failed to create server-running pod", "name", serverRunningPod.Name)
+		return err, true
+	}
+	logger.V(5).Info("Created server-running pod", "name", serverRunningPod.Name)
+
+	logger.V(5).Info("Processed server-requesting pod", "name", requestingPod.Name)
+	return nil, false
+}
+
+func composeServerRunningPod(reqPod *corev1.Pod, data api.RunnerData) (*corev1.Pod, error) {
+	rawTmpl, ok := reqPod.Annotations[api.ServerPatchAnnotationName]
+	if !ok {
+		return nil, fmt.Errorf("annotation %q not found", api.ServerPatchAnnotationName)
+	}
+
+	tmpl, err := template.New("serverPatch").Option("missingkey=error").Parse(rawTmpl)
+	if err != nil {
+		return nil, fmt.Errorf("parse template: %w", err)
+	}
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, data); err != nil {
+		return nil, fmt.Errorf("execute template: %w", err)
+	}
+	renderedPatch := buf.Bytes()
+
+	patchJSON, err := yaml.YAMLToJSON(renderedPatch)
+	if err != nil {
+		return nil, fmt.Errorf("yaml to json: %w", err)
+	}
+
+	// remove the requester container
+	for i, c := range reqPod.Spec.Containers {
```

Comment:
https://pkg.go.dev/slices@go1.24.2#DeleteFunc is a thing too

Reply:
Now that we use the same name for the two containers, so this removal logic is not need anymore and I removed the removal logic.